### ISSUE TITLE
Navigating to the root ("/") dosen't work in the app router when hosted on Github pages

### ISSUE THREAD

**Issue by mohamednehad450** [vercel/next.js#51549]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
    Platform: linux
    Arch: x64
    Version: #46-Ubuntu SMP Mon May 22 15:18:07 UTC 2023
Binaries:
    Node: 16.20.0
    npm: 8.19.4
    Yarn: 1.22.19
    pnpm: N/A
Relevant packages:
    next: 13.4.7-canary.1
    eslint-config-next: N/A
    react: 18.2.0
    react-dom: 18.2.0
    typescript: 4.9.5
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/mohamednehad450/repoduction-app-dir

### To Reproduce

1. go to [the reproduction repo hosted on GitHub pages](https://mohamednehad450.github.io/repoduction-app-dir/)
2. click on "/test"
3. click on the first link (<Link href={"/"}>), notice the href is set correctly to "https://mohamednehad450.github.io/repoduction-app-dir/"

it navigates to a partial url of the root (https://mohamednehad450.github.io/repoduction-app), not the complete one (https://mohamednehad450.github.io/repoduction-app-dir/).

### Describe the Bug

When using The Link Element to navigate to the root ("/"), the href is set correctly, however, when clicked it navigates to a paritial link of the href, it seems to remove the last 4 characters from the link

the issue is also present when using  the `useRouter` hook from `next/navigation`

### Expected Behavior

When using the Link tag to navigate to the root ("/"), it should navigate to the project root not some partial link

### How are you deploying your application? (if relevant)

Github Pages

**Comment by babytutu** [comment]
Open next.config.js and add the trailingSlash config:

module.exports = {
  trailingSlash: true,
}

With this option set, urls like /about will redirect to /about/.

https://nextjs.org/docs/app/api-reference/next-config-js/trailingSlash

**Comment by mohamednehad450** [comment]
Adding the trailingSlash fixed the issue

however I do think its still a bug since it trims 4 characters off the url

 also this is not an issue when using the pages router, here is the same example but with the pages router [repo](https://github.com/mohamednehad450/reproduction-pages) and [the live site](https://mohamednehad450.github.io/reproduction-pages)

**Comment by Naes0** [comment]
Not sure if it helps anyone but I fixed it by enabling trailingSlash and also setting prefetch={false} on the link going to '/'.

**Comment by icyJoseph** [comment]
A fix is incoming! https://github.com/vercel/next.js/commit/539ae6c1fc39293633d0835afb73e483d2c2f7ed using trailing slashes should not be needed!

**Comment by icyJoseph** [comment]
This has been fix in `next@13.4.11` - this no longer removes 4 characters of the URL :)

**Comment by mohamednehad450** [comment]
> This has been fix in `next@13.4.11` - this no longer removes 4 characters of the URL :) [#51845 (comment)](https://github.com/vercel/next.js/issues/51845#issuecomment-1645161990)
